egs/librispeech/ASR/conformer_ctc/pretrained.py

```diff
@@ -103,28 +132,39 @@ def main():
     checkpoint = torch.load(args.checkpoint, map_location="cpu")
     model.load_state_dict(checkpoint["model"])
     model.to(device)
+    model.eval()
 
     HLG = k2.Fsa.from_dict(torch.load(params.hlg))
     HLG = HLG.to(device)
 
-    model.to(device)
+    opts = kaldifeat.FbankOptions()
+    opts.device = device
+    opts.frame_opts.dither = 0
+    opts.frame_opts.snip_edges = False
+    opts.frame_opts.samp_freq = params.sample_rate
+    opts.mel_opts.num_bins = params.feature_dim
 
-    wave, sample_freq = torchaudio.load(params.sound_file)
-    assert sample_freq == params.sample_freq
-    wave = wave.to(device)
+    fbank = kaldifeat.Fbank(opts)
 
-    features = torchaudio.compliance.kaldi.fbank(
-        waveform=wave,
-        num_mel_bins=params.feature_dim,
-        snip_edges=False,
-        sample_frequency=params.sample_freq,
+    waves = read_sound_files(
+        filenames=params.sound_files, expected_sample_rate=params.sample_rate
     )
+    waves = [w.to(device) for w in waves]
 
-    features = features.unsqueeze(0)
+    logging.info(f"Decoding started")
+    features = fbank(waves)
```

Comment:
Replacing torchaudio.compliance.kaldi with kaldifeat
since it is easier to extract features for multiple sound
files at the same time.

Reply:
Nice. I still have adding kaldifeat to Lhotse on my radar. I might remove all other kaldi-related feature extractors at the same time. But I think I won’t be able to do it before the tutorial.